x/tokenfactory/keeper/before_send.go

```diff
@@ -0,0 +1,155 @@
+package keeper
+
+import (
+	"encoding/json"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/neutron-org/neutron/x/tokenfactory/types"
+
+	errorsmod "cosmossdk.io/errors"
+	wasmvmtypes "github.com/CosmWasm/wasmvm/types"
+)
+
+func (k Keeper) setBeforeSendHook(ctx sdk.Context, denom string, cosmwasmAddress string) error {
+	// verify that denom is an x/tokenfactory denom
+	_, _, err := types.DeconstructDenom(denom)
+	if err != nil {
+		return err
+	}
+
+	store := k.GetDenomPrefixStore(ctx, denom)
+
+	// delete the store for denom prefix store when cosmwasm address is nil
+	if cosmwasmAddress == "" {
+		store.Delete([]byte(types.BeforeSendHookAddressPrefixKey))
+		return nil
+	}
+
+	_, err = sdk.AccAddressFromBech32(cosmwasmAddress)
+	if err != nil {
+		return err
+	}
+
+	store.Set([]byte(types.BeforeSendHookAddressPrefixKey), []byte(cosmwasmAddress))
+
+	return nil
+}
+
+func (k Keeper) GetBeforeSendHook(ctx sdk.Context, denom string) string {
+	store := k.GetDenomPrefixStore(ctx, denom)
+
+	bz := store.Get([]byte(types.BeforeSendHookAddressPrefixKey))
+	if bz == nil {
+		return ""
+	}
+
+	return string(bz)
+}
+
+func CWCoinsFromSDKCoins(in sdk.Coins) wasmvmtypes.Coins {
```

Comment:
unused func

Reply:
This is the same as in original `osmosis` module